src/zeit.zig

```diff
@@ -462,6 +462,71 @@ pub const Date = struct {
     year: i32,
     month: Month,
     day: u5, // 1-31
+
+    /// Checks for equality of two dates
+    pub fn eql(date1: @This(), date2: @This()) bool {
+        return date1.year == date2.year and
+               date1.month == date2.month and
+               date1.day == date2.day;
+    }
+
+    test "Date-Equality" {
+        const date = Date{
+            .year = 2025,
+            .month = Month.sep,
+            .day = 13,
+        };
+        try std.testing.expect(date.eql(Date{ .year = 2025, .month = Month.sep, .day = 13 }));
+        try std.testing.expect(!date.eql(Date{ .year = 2025, .month = Month.sep, .day = 12 }));
+        try std.testing.expect(!date.eql(Date{ .year = 2025, .month = Month.aug, .day = 13 }));
+        try std.testing.expect(!date.eql(Date{ .year = 2024, .month = Month.sep, .day = 13 }));
+    }
+
+    /// Compares two dates with another. If `date2` happens after `date1`, then the `TimeComparison.after` is returned.
+    /// If `date2` happens before `date1`, then `TimeComparison.before` is returned. If both represent the same date, `TimeComparison.equal` is returned;
+    pub fn compare(date1: @This(), date2: @This()) TimeComparison {
+        if(date1.year > date2.year) {
```

Comment:
I think this would be cleaner to do:

```
const time1: Time = .{
    .year = date1.year,
    .month= date1.month,
    .day = date1.day
};
const time2 = .{ ... };

const inst1 = instant(.{ .time = time1 });
const inst2 = ...;

if (inst1.timestamp > inst2.timestamp) 
    return .before
else if (inst1.timestamp < inst2.timestamp)
    return .after
else
    return .equal;
```

What do you think?

Reply:
You know what, this calls into `daysFromCivil` which will probably not be faster even though it has less branching. Let's go with what you have.